tasks/ThrusterTask.cpp

```diff
@@ -100,11 +100,11 @@ void ThrusterTask::setGazeboModel( std::string const& pluginName, ModelPtr model
 {
     string worldName = GzGet((*(model->GetWorld())), Name, ());
 
-    string taskName = "gazebo::" + worldName + "::" + model->GetName() + "::" + pluginName;
+    string taskName = std::regex_replace(pluginName, std::regex("__"), "::");
     provides()->setName(taskName);
     _name.set(taskName);
 
-    topicName = model->GetName() + "/thrusters";
+    string topicName = std::regex_replace(pluginName, std::regex("__"), "/") + "/thrusters";
```

Comment:
You redeclare topicName as a local variable. This statement does nothing (just drop the "string")

(same comment applies to all the other changes)